sites/all/modules/custom/dccr_blocks/dccr_blocks.module

```diff
@@ -369,3 +387,278 @@ function dccr_blocks_create_account_block() {
 
   return $block;
 }
+
+/**
+ * Get Final Schedule block.
+ */
+function dccr_get_final_schedule($only_flagged, $empty_slot_text, $block_title) {
+  $block = array();
+  $markup = '';
+  $terms = dccr_get_rooms();
+  $tracks = taxonomy_get_tree(4);
+  $terms_count = 0;
+  $counter = 0;
+  $params = drupal_get_query_parameters();
+  $markup .= '<div class=\'tracks\'>';
+  for ($i = 0; $i < count($tracks); $i++) {
+    $markup .= '<div class=\'color-' . strtolower($tracks[$i]->name) . '\'><p> ' . $tracks[$i]->name . '</p></div>';
+  }
+  $markup .= '</div><div class=\'filter\'><div class=\'day\'>' . l(t('July 30th'), 'schedule', array('query' => array('date' => 'thursday'))) . '</div><div class=\'day\'>' . l(t('July 31th'), 'schedule', array('query' => array('date' => 'friday'))) . '</div></div>';
+
+  if ($only_flagged) {
+    drupal_add_js(drupal_get_path('module', 'dccr_blocks') . '/js/alter_flag_behavior.js');
+  }
+
+  // Thursday, July 30th.
+  $date1 = 1438264800;
+  // Friday, July 31st.
+  $date2 = 1438351200;
+  if (isset($params['date']) && check_plain($params['date'])) {
+    if ($params['date'] == 'friday') {
+      // Friday, July 31st.
+      $date1 = 1438351200;
+      // Saturday, August 1st.
+      $date2 = 1438437600;
+    }
+  }
+  $activities = dccr_get_global_activities(array($date1, $date2));
+
+  // Create markup for rooms.
+  $markup .= '<div class=\'rooms\'>';
+  foreach ($terms as $term) {
+    $terms_count++;
+    $markup .= '<div class=\'room-name\'><h2>' . $term->name . '</h2></div>';
```

Comment:
I think this is not necessary with rooms but it is already implemented with tracks.